event-service-rest/src/main/java/com/restapi/rest/MainController.java

```diff
@@ -1,13 +1,141 @@
 package com.restapi.rest;
 
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RestController;
+import com.restapi.api.EventService;
+import com.restapi.dto.Event;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
 
 @RestController
+@RequestMapping("/api/v1")
 public class MainController {
 
-    @GetMapping("/test")
-    String getTest() {
-        return "This is a test REST response";
+    private EventService eventService;
+
+    @Autowired
+    public MainController(EventService eventService) {
+        this.eventService = eventService;
+    }
+
+    /**
+     * Create new event
+     * @param payload
+     * @return
+     */
+    @PostMapping("/events")
+    public ResponseEntity<EventApi> createEvent(@RequestBody EventApi payload) {
+
+        Event event = payload.getEvent();
+
+        event = eventService.createEvent(event);
+
+        return new ResponseEntity<>(EventApi.builder()
+                .id(event.getId())
+                .title(event.getTitle())
+                .eventType(event.getEventType())
+                .place(event.getPlace())
+                .speaker(event.getSpeaker())
+                .dateTime(event.getDateTime())
+                .build(),
+                HttpStatus.CREATED);
+    }
+
+    /**
+     * Get all events from the storage.
+     * @return
+     */
+    @GetMapping("/events")
+    public ResponseEntity<List<EventApi>> getEventList() {
+
+        List<EventApi> events = eventService.getAllEvents()
+                .stream().map(e -> new EventApi(e.getId(),
+                        e.getTitle(),
+                        e.getPlace(),
+                        e.getSpeaker(),
+                        e.getEventType(),
+                        e.getDateTime()))
+                .collect(Collectors.toList());
+
+        if (events.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        }
+
+        return new ResponseEntity<>(events,
+                HttpStatus.OK);
+    }
+
+    /**
+     * Get one event by id
+     * @param id
+     * @return
+     */
+    @GetMapping("/events/{id}")
+    public ResponseEntity<EventApi> getById(@PathVariable Integer id) {
+
+        Event event = eventService.getEvent(id);
+
+        if (event == null) {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
```

Comment:
it depends on the requirements of course but usually not finding an entity returns 404; that's what I have seen on all my production projects
https://stackoverflow.com/questions/26845631/is-it-correct-to-return-404-when-a-rest-resource-is-not-found

Reply:
it would be useful to return a message to the client as well

Reply:
> it may be useful to return a message to the client as well but depends on the requirements

done